File: EveVoid/Dto/SignatureDto.cs
using EveVoid.Models.Navigation;
using System;
using System.Collections.Generic;

namespace EveVoid.Dto
{
    public class SignatureDto
    {
        public int Id { get; set; }
        public int SystemId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string SignatureId { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime ExpiryDate { get; set; }
        public DateTime LastUpdate { get; set; }
        public SignatureType SignatureType { get; set; }

        //wormhole
        public int? WormholeTypeId { get; set; }
        public MassIndicator MassIndicator { get; set; }
        public TimeRemainingIndicator TimeRemainingIndicator { get; set; }
        public int? DestinationSystemId { get; set; }
        public string DestinationSystemName { get; set; }
        public string DestinationWormholeType { get; set; }
        public double TotalMass { get; set; }
        public string WormholeType { get; set; }

        public List<JumpDto> Jumps { get; set; }

        public SignatureDto()
        {
            MassIndicator = MassIndicator.Unset;
            TimeRemainingIndicator = TimeRemainingIndicator.Unset;
            Jumps = new List<JumpDto>();
        }
    }
}

File: EveVoid/ClientApp/src/app/api/models/signature-dto.ts
/* tslint:disable */
import { JumpDto } from './jump-dto';
import { MassIndicator } from './mass-indicator';
import { SignatureType } from './signature-type';
import { TimeRemainingIndicator } from './time-remaining-indicator';
export interface SignatureDto {
  creationDate?: string;
  description?: string;
  destinationSystemId?: number;
  destinationSystemName?: string;
  destinationWormholeType?: string;
  expiryDate?: string;
  id?: number;
  jumps?: Array<JumpDto>;
  lastUpdate?: string;
  massIndicator?: MassIndicator;
  name?: string;
  signatureId?: string;
  signatureType?: SignatureType;
  systemId?: number;
  timeRemainingIndicator?: TimeRemainingIndicator;
  totalMass?: number;
  wormholeType?: string;
  wormholeTypeId?: number;
}
